05.wc/wc.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = parse_options
+  if ARGV.empty?
+    handle_stdin(options)
+  else
+    handle_files(ARGV, options)
+  end
+end
+
+def parse_options
+  options = { lines: false, words: false, bytes: false }
+  OptionParser.new do |opts|
+    opts.banner = 'Usage: wc.rb [options] [file ...]'
+    opts.on('-l', 'Count lines') { options[:lines] = true }
+    opts.on('-w', 'Count words') { options[:words] = true }
+    opts.on('-c', 'Count bytes') { options[:bytes] = true }
+  end.parse!
+  options.values.any? ? options : options.transform_values { true }
+end
+
+def handle_stdin(options)
+  file_content = $stdin.read
+  output_counts_hash = { counts: count_all(file_content) }
+  output_results(output_counts_hash, options)
+end
+
+def handle_files(files, options)
+  file_counts = files.map { |file| count_file_contents(file) }
+  file_counts.each { |file_data| output_results(file_data, options) }
+  output_total(file_counts, options) if files.size > 1
+end
+
+def output_total(file_counts, options)
+  total_counts = calculate_totals(file_counts)
+  output_content_hash = { counts: total_counts, filename: 'total' }
+  output_results(output_content_hash, options)
+end
+
+def count_file_contents(file)
+  file_content = File.read(file)
+  { counts: count_all(file_content), filename: file }
+end
+
+def calculate_totals(file_counts)
+  total_counts = { lines: 0, words: 0, bytes: 0 }
+  file_counts.each do |data|
+    total_counts[:lines] += data[:counts][:lines]
+    total_counts[:words] += data[:counts][:words]
+    total_counts[:bytes] += data[:counts][:bytes]
+  end
+  total_counts
+end
+
+def count_all(content)
+  {
+    lines: content.count("\n"),
+    words: content.split(/\s+/).reject(&:empty?).count,
+    bytes: content.bytesize
+  }
+end
+
+def construct_output_count_values(file_info, options)
+  file_info[:counts].keys.filter_map do |key|
+    options[key] || options.values.none? ? file_info[:counts][key].to_s.rjust(8) : nil
```

Comment:
```suggestion
    file_info[:counts][key].to_s.rjust(8) if options[key] || options.values.none?
```

好みの問題ですがこっちの方がわずかに短くなりますね

Reply:
@JunichiIto 
ありがとうございます。三項演算子文より可読性も上がって良いなと感じました！